Scryfall Tools/limited_pools.py

```diff
@@ -4,105 +4,103 @@
-        if 'Legendary' in card['type_line']:
-            # If the card is legendary, replace it with the etched foil frame 50% of the time
-            etched_foil_rng = random.randint(0, 1)
-            if etched_foil_rng == 1:
-                # Replace the card with the foil etched version of the same name
-                card = st.search_for_cards(f's:cmr {card_name} frame:etched')[0]
-        
+
+        # There is a 50% chance a legendary foil
+        # will be replaced with a foil etched version.
+        if 'Legendary' in card['type_line'] and \
+                'Creature' in card['type_line']:
+            if random.randint(0, 1) == 1:
+                card = st.search_for_cards(
+                    f's:cmr {card["name"]} frame:etched'
+                )[0]
+
         pack += [card]
-        
-        # Get 2 legendary cards of any rarity
-        legendary_rarity_1 = random.choices(
+
+        # Add 2 legendary cards of any rarity
+        rarities = random.choices(
             ['m', 'r', 'u'],
-            [1, 8, 23]
+            [1, 8, 23],
+            k=2
         )
-        legendary_rarity_2 = random.choices(
-            ['m', 'r', 'u'],
-            [1, 8, 23]
+
+        legendary_card_1 = st.get_random_card(
+            f'set:{set_code} t:legendary is:booster r:{rarities[0]}'
         )
-        
-        legendary_card_1 = st.get_random_card(f'set:{set_code} t:legendary is:booster r:{legendary_rarity_1}')
-        pack += [legendary_card_1]
-        legendary_card_1_name = legendary_card_1['name']
-        pack += [st.get_random_card(f'set:{set_code} t:legendary is:booster -"{legendary_card_1_name}" r:{legendary_rarity_2}')]
-        
-        # Get a rare or mythic (~14% chance of Mythic)
-        if random.randint(0, 7) == 7:
-            rare_card = st.get_random_card(f'set:{set_code} r=m is:booster -t:legendary')
-        else:
-            rare_card = st.get_random_card(f'set:{set_code} r=r is:booster -t:legendary')
-
-        pack += [rare_card]
-        
-        # Get 3 uncommons
-        cards = st.search_for_cards(f'set:{set_code} r=u -t:legendary is:booster')
-        uncommons = random.sample(list(cards), 3)
-        pack += uncommons
-        
-        # Get 13 commons
-        # Determine if Prismatic piper replaces one of the foils
+        pack += [
+            legendary_card_1,
+            st.get_random_card(
+                f'set:{set_code} t:legendary is:booster '
+                f'-"{legendary_card_1["name"]}" '
+                f'r:{rarities[1]}'
+            )
+        ]
+
+        # Add a rare or mythic (~14% chance of Mythic)
+        pack += [st.get_random_card(
+            f'set:{set_code} is:booster '
+            f'r={random.choices(["m", "r"], [1, 6])[0]}'
+        )]
+
+        # Add 3 uncommons
+        cards = st.search_for_cards(
+            f'set:{set_code} r=u -t:legendary is:booster'
+        )
+        pack += [random.sample(list(cards), 3)]
```

Comment:
This leads to a list in the middle of the pack; as with the commons, no brackets are needed.

Reply:
```suggestion
        pack += random.sample(list(cards), 3)
```